sdk/program/src/system_instruction.rs

```diff
@@ -450,6 +961,175 @@ pub fn transfer_with_seed(
     )
 }
 
+/// Allocate space for an account.
+///
+/// This function produces an [`Instruction`] which must be submitted in a
+/// [`Transaction`] or [invoked] to take effect, containing a serialized
+/// [`SystemInstruction::Allocate`].
+///
+/// [`Transaction`]: https://docs.rs/solana-sdk/latest/solana_sdk/transaction/struct.Transaction.html
+/// [invoked]: crate::program::invoke
+///
+/// Account creation typically involves three steps: [`allocate`] space,
+/// [`transfer`] lamports for rent, [`assign`] to its owning program. The
+/// [`create_account`] function does all three at once.
```

Comment:
Remove this paragraph?

Reply:
Also, should we mention that in order for the instruction to succeed, the account must have empty data? And maybe that the requested data size must be less than `MAX_PERMITTED_DATA_LENGTH`?

Reply:
I'll mention those.

Reply:
@CriesofCarrots I've seen a lot of code relating to resizing accounts, and am not sure whether that is possible yet or not, and whether it is done through this system instruction or not. Can you clarify?

Reply:
Sorry for letting this question fall through the cracks! Yes, resizing accounts is live on mainnet-beta. The easiest way to reallocate is [this method on AccountInfo](https://github.com/solana-labs/solana/blob/a422387ca83bf309b8230aefadac8dcbd73314c6/sdk/program/src/account_info.rs#L147), but I think that it would technically be possible to do it with this system instruction, if the program had previously emptied the account.

Reply:
I'm happy to leave it out!